design/one-pager-simple-class-selection.md

```diff
@@ -0,0 +1,411 @@
+can be onerous.
+
+## Terminology
+
+Most terminology is covered by the [Crossplane glossary]. Key terms used in this
+document include:
+
+* Crossplane uses a _provider_ to satisfy infrastructure needs. Google Cloud
+  Platform (GCP), Elastic Cloud, and Kubernetes are all providers.
+* A resource is said to be _portable_ if it is not tightly coupled to any one
+  provider. Managed resources like `RDSInstance` are coupled to a provider
+  because they expose provider specific configuration parameters, whereas
+  resource claims like `PostgreSQLInstance` are portable because they do not.
+* _Dynamic provisioning_  is the act of satisfying a _portable_ request for
+  infrastructure (i.e. a resource claim) by automatically provisioning a
+  non-portable managed resource.
+* _Resource classes_ enable dynamic provisioning. A resource class determines
+  what class of managed resource should be created to satisfy a resource claim.
+* _Separation of concern_ happens when an opinionated party can publish classes of
+  infrastructure to be consumed by parties unconcerned with such details.
+  Imagine an organisation in which an infrastructure team supports many product
+  teams who own the lifecycle of their infrastructure. Separation of concerns
+  allows the infrastructure team to define a 'production database' class of
+  service that happens to be a high availability, SSD backed GCP CloudSQL
+  instance. Product teams can then self-service by provisioning a production
+  database and trust that it will be configured appropriately.
+* Crossplane recommends using Kubernetes namespaces to logically group
+  infrastructure into an _infrastructure namespace_, for example
+  `production-infra` or `us-east-coast`. This infrastructure could be claimed
+  and used by applications running in a distinct _application namespace_ such as
+  `widget-service` or `acme-team`.
+
+Note that two forms (classes?) of resource class currently exist; see [portable
+resource classes] for details.
+
+## Goals
+
+The goal of this proposal is to __make it easier for potential Crossplane users
+to learn and take advantage of the portability and separation of concerns__ that
+are enabled by dynamic provisioning. A _secondary_ goal is to maintain the
+flexibility and power that Crossplane offers complex organisations with many
+application and infrastructure namespaces.
+
+## Proposal
+
+This document proposes Crossplane retire the concept of portable resource
+classes. Resource claims would instead be matched directly to non-portable
+resource classes (referred to henceforth as simply 'resource classes') using
+[label selectors]. Providers, managed resources, and resource classes would be
+cluster scoped rather than namespaced.
+
+```yaml
+apiVersion: database.crossplane.io/v1alpha1
+kind: PostgreSQLInstance
+metadata:
+  namespace: acme-team
+  name: gitlab-database
+spec:
+  classSelector:
+    matchLabels:
+      stack: gitlab
```

Comment:
does the `stack:gitlab` label  example recommend a pattern for stack-specific, cluster-scoped resource classes be created for each type of stack in the cluster?

Reply:
That's what I was thinking, though I'd frame it more as an example of a potential pattern or maybe a _speculative_ recommendation.

I could see this either working as:

* The Gitlab stack installs a beginner resource class (or two) that the resource claims  produced by new `GitLab` resources match by default.
* The Gitlab stack documents that new `GitLab` instances will create resource claims that match on something like these labels, and it's up to the user to make sure there are classes that would match these labels?

Not sure how prescriptive we should be about this, but it would at least open the possibility to stack authors.